zerver/tests/test_subs.py

```diff
@@ -2198,6 +2204,86 @@ def test_unarchive_stream(self) -> None:
             },
         )
 
+    def test_permissions_and_archiving_behavior(self) -> None:
+        """
+        Test permissions for archiving and unarchiving streams, and ensure users without
+        the necessary permissions cannot archive or unarchive a stream.
+        """
+        desdemona = self.example_user("desdemona")
+
+        iago = self.example_user("iago")
+
+        self.login_user(desdemona)
+        stream = self.make_stream("test_stream", invite_only=False)
+        self.subscribe(iago, stream.name)
+
+        result = self.client_delete(f"/json/streams/{stream.id}")
+        self.assert_json_success(result)
+        stream.refresh_from_db()
+        self.assertTrue(stream.deactivated)
+
+        result = self.client_patch(
+            f"/json/streams/{stream.id}", {"is_archived": orjson.dumps(False).decode()}
+        )
+        self.assert_json_success(result)
+
+        cordellia = self.example_user("cordelia")
+        self.login_user(cordellia)
+
+        stream1 = self.make_stream("test_stream_1", invite_only=False)
+        self.subscribe(iago, stream1.name)
+
+        result = self.client_delete(f"/json/streams/{stream.id}")
+        self.assert_json_error(result, "You do not have permission to administer this channel.")
+        stream1.refresh_from_db()
+        self.assertTrue(stream.deactivated)
+
+        result = self.client_patch(
+            f"/json/streams/{stream1.id}", {"is_archived": orjson.dumps(False).decode()}
+        )
+        self.assert_json_error(result, "You do not have permission to administer this channel.")
+
+        aaron = self.example_user("aaron")
+        self.login_user(aaron)
+
+        self.subscribe(aaron, stream.name)
+
+        result = self.client_delete(f"/json/streams/{stream.id}")
+        self.assert_json_error(result, "You do not have permission to administer this channel.")
+
+        result = self.client_patch(
```

Comment:
Why have you removed the check for making sure that user with channel administering permission can unarchive the stream?